packages/xo-server/src/xo-mixins/proxies.js

```diff
@@ -329,19 +329,15 @@ export default class Proxy {
 
   async callProxyMethod(id, method, params, expectStream = false) {
     const proxy = await this._getProxy(id)
-    if (proxy.address === undefined) {
-      if (proxy.vmUuid === undefined) {
-        throw new Error(
-          'proxy VM and proxy address should not be both undefined'
-        )
-      }
+    if (proxy.address === undefined && proxy.vmUuid === undefined) {
+      throw new Error('proxy VM and proxy address should not be both undefined')
+    }
 
+    if (proxy.vmUuid !== undefined) {
       const vm = this._app.getXapi(proxy.vmUuid).getObjectByUuid(proxy.vmUuid)
-      if (
-        (proxy.address = extractIpFromVmNetworks(
-          vm.$guest_metrics?.networks
-        )) === undefined
-      ) {
+      proxy.address =
```

Comment:
Don't mutate the `proxy` object.

Reply:
It's impossible that the same reference will be used in multiple place.